File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        output = []
        record = {}
        for i in range(len(nums)):
            num1 = target - nums[i]
            if num1 in record:
                output.append(record[num1])
                output.append(i)
            else:
                record[nums[i]] = i
        return output 

            
            

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l = 0, r = numbers.size() - 1;
        while(l < r){
            int current = numbers[l] + numbers[r];
            if(current < target){
                l++;
            }else if(current > target){
                r--;
            }else{
                return {l + 1, r + 1};
            }
        }
        return {};
    }
};
